docs/userguide/annotations.md

```diff
@@ -1,65 +1,68 @@
 
-  -- %suitesetup
+  -- %beforeall
   procedure globalsetup;
 
-  -- %suiteteardown
+  -- %afterall
   procedure global_teardown;
 
   /* Such comments are allowed */
 
-  -- %test(Name of test1)
-  -- %testtype(smoke)
+  -- %test
+  -- %displayname(Name of test1)
   procedure test1;
 
-  -- %test(Name of test2)
-  -- %testsetup(setup_test1)
-  -- %testteardown(teardown_test1)
+  -- %test
+  -- %displayname(Name of test2)
+  -- %beforetest(setup_test1)
+  -- %aftertest(teardown_test1)
   procedure test2;
 
-  -- %test(Name of test3)
+  -- %test
+  -- %displayname(Name of test3)
   -- %testtype(smoke)
   procedure test3;
 
   procedure setup_test1;
 
   procedure teardown_test1;
 
-  -- %setup
+  -- %beforeeach
   procedure setup;
 
-  -- %teardown
+  -- %aftereach
   procedure teardown;
 
 end test_pkg;
 ```
 
 #Annotations meaning
 
-Annotation | Meaning
------------- | -------------
-%suite | Marks package to be a suite with it procedures as tests. This way all testing packages might be found in the schema. Parameter of the annotation is the Suite name
-%suitepackage | Similar to java package. The example suite should be put as an element of the "globaltests" suite which is an element of the "all" suite. This allows one to execute "glovaltests" suite which would recursively run all the child suites including this one.
-%suitetype | The way for suite to have something like a type. One might collect suites based on the subject of tests (a system module for example). There might be critical tests to run every time and more covering but slow tests. This technique allows to configure something like "fast" testing.
-%setup | Marks procedure as a default setup procedure for the suite.
-%teardown | Marks procedure as a default teardown procedure for the suite.
-%test | Marks procedure as a test. Parameter is a name of the test
-%testtype | Another way to tag tests to filter afterwards
-%testsetup | Marks that special setup procedure has to be run before the test instead of the default one
-%testteardown | Marks that special teardown procedure has to be run before the test instead of the default one
-%suitesetup | Procedure with executes once at the beginning of the suite and doesn't executes before each test
-%suiteteardown | Procedure with executes once after the execution of the last test in the suite.
+| Annotation |Level| Describtion |
+| --- | --- | --- |
+| `%suite` | Package | Marks package to be a suite of tests This way all testing packages might be found in a schema. |
+| `%suitepath(<path>)` | Package | Similar to java package. The annotation allows logical grouping of suites into hierarcies. |
+| `%displayname(<description>)` | Package/procedure | Human-familiar describtion of the suite/test. `%displayname(Name of the suite/test)` |
+| `%test` | Procedure | Denotes that a method is a test method. |
+| `%beforeall` | Procedure | Denotes that the annotated procedure should be executed before all `%test` methods in the current suite. |
```

Comment:
Denotes that the annotated procedure should be executed **before all elements of the current suite**.
If `%beforeall` is defined more then once, the last `%beforeall` in the suite is used.